x/oracle/keeper.go

```diff
@@ -0,0 +1,364 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	gogotypes "github.com/gogo/protobuf/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	paramstypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/umee-network/umee/x/oracle/types"
+)
+
+// Keeper of the oracle store
+type Keeper struct {
+	cdc        codec.BinaryCodec
+	storeKey   sdk.StoreKey
+	paramSpace paramstypes.Subspace
+
+	accountKeeper types.AccountKeeper
+	bankKeeper    types.BankKeeper
+	distrKeeper   types.DistributionKeeper
+	StakingKeeper types.StakingKeeper
+
+	distrName string
+}
+
+// NewKeeper constructs a new keeper for oracle
+func NewKeeper(cdc codec.BinaryCodec, storeKey sdk.StoreKey,
+	paramspace paramstypes.Subspace, accountKeeper types.AccountKeeper,
+	bankKeeper types.BankKeeper, distrKeeper types.DistributionKeeper,
+	stakingKeeper types.StakingKeeper, distrName string) Keeper {
+
+	// ensure oracle module account is set
+	if addr := accountKeeper.GetModuleAddress(types.ModuleName); addr == nil {
+		panic(fmt.Sprintf("%s module account has not been set", types.ModuleName))
+	}
+
+	// set KeyTable if it has not already been set
+	if !paramspace.HasKeyTable() {
+		paramspace = paramspace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	return Keeper{
+		cdc:           cdc,
+		storeKey:      storeKey,
+		paramSpace:    paramspace,
+		accountKeeper: accountKeeper,
+		bankKeeper:    bankKeeper,
+		distrKeeper:   distrKeeper,
+		StakingKeeper: stakingKeeper,
+		distrName:     distrName,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+//-----------------------------------
+// ExchangeRate logic
+
+// GetExchangeRate gets the consensus exchange rate of USD denominated in the denom asset from the store.
```

Comment:
Does that mean the value returned will be the inverse of the token's USD value?

e.g. If 1 `MISC` is worth 3 `USD`, then would `GetExchangeRate(ctx, "MISC")` return 0.33?

Reply:
Great question, that is my understanding as the rates are represented in the denom of the asset in the map[].